### PR TITLE
Add new stable version to Microsoft.Security : 2018-06-01 with pricing configuration

### PR DESCRIPTION
**Pull request by amireluk**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by sergey-shandar — CHANGES_REQUESTED** [review]
@amireluk, please fix this syntax error:
```
Error resolving $ref pointer "/home/travis/build/Azure/azure-rest-api-specs/specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json#/definitions/Resource"
```

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+            "description": "Pricing tier will be applied for the scope based on the resource ID",
+            "properties": {
+                "properties": {
+                    "x-ms-client-flatten": true,
+                    "description": "Pricing data",
+                    "$ref": "#/definitions/PricingProperties"
+                }
+            },
+            "allOf": [{
+                "$ref": "#/definitions/Resource"
```
Comment (chlahav) [inline specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: this definition is missing (this is what is failing the test)

Comment (amireluk) [reply specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: thanks

**Comment by amireluk** [comment]
@sergey-shandar - I've addressed your request with commit 911838c

**Review by sergey-shandar — APPROVED** [review]
We still need ARM approval

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+            "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+            "flow": "implicit",
+            "description": "Azure Active Directory OAuth2 Flow",
+            "scopes": {
+                "user_impersonation": "impersonate your user account"
+            }
+        }
+    },
+    "paths": {
+        "/subscriptions/{subscriptionId}/providers/Microsoft.Security/pricings": {
```
Comment (simongdavies) [inline specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: The name pricings seems a little strange, based on the description would pricingConfigurations be better?

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+            "get": {
+                "x-ms-examples": {
+                    "Get pricings on subscription": {
+                        "$ref": "./examples/Pricings/ListPricings_example.json"
+                    }
+                },
+                "tags": ["Pricings"],
```
Comment (simongdavies) [inline specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: should this support paging?

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+                        }
+                    }
+                }
+            },
+            "put": {
+                "x-ms-examples": {
+                    "Update pricing on subscription": {
+                        "$ref": "./examples/Pricings/PutPricingByName_example.json"
+                    }
+                },
```
Comment (simongdavies) [inline specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: Do you need to add x-ms-long-running-operation": true?

**Review by simongdavies — CHANGES_REQUESTED** [review]

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json (hunk 2 above)
Comment (amireluk) [reply specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: If you're referring to the resource name, it's the same one used in previous versions, I don't want to create a new resource name.

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json (hunk 3 above)
Comment (amireluk) [reply specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: Probably not.
It's limited to a closed list of pricing configurations. There are currently 3, expected to grow slightly but not by a lot.

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json (hunk 4 above)
Comment (amireluk) [reply specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: Current implementation does not expect it.
should it ?

**Comment by sergey-shandar** [comment]
@simonjaeger any progress on this one?

File: specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json (hunk 4 above)
Comment (simongdavies) [reply specification/security/resource-manager/Microsoft.Security/stable/2018-06-01/security.json]: The swagger shows the put as a long running operation that returns a 202, in which case it should have a location header that contains an operation url that can be polled for the status of the put, if that is the case then adding x-ms-long-running-operation will cause code to be generated to support this, if the operation is not async then it should return 200 0r 201 (https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#response)

**Comment by sergey-shandar** [comment]
@amireluk are you working on the PR?

**Comment by sergey-shandar** [comment]
Closing the PR. @amireluk let me know if you need to reopen it.

**Comment by amireluk** [comment]
@sergey-shandar please reopen the CR.
I want to push new changes.

**Pull request closed without merging**